src/mame/drivers/blit.cpp

```diff
@@ -0,0 +1,312 @@
+	bool m_sysram_on;
+
+	uint8_t *m_videoram;
+	uint32_t m_videostart = 0; // 0x27000;
+};
+
+void blit_state::blit_mem(address_map &map)
+{
+	map.unmap_value_high();
+	map(0x000000, 0x000007).r(FUNC(blit_state::bootvect_r));   /* ROM mirror just during reset */
+	map(0x000000, 0x000007).w(FUNC(blit_state::bootvect_w));   /* After first write we act as RAM */
+	map(0x000000, 0x03ffff).ram().share("p_ram");
+	map(0x040000, 0x045fff).rom().region(M68K_TAG, 0);
+	// octal 0000, 0002 - 16-bit
+//	map(0x060000, 0x060003).r(FUNC(blit_state::mouse_xy));
+	// octal 0011, 0013 - 8-bit - host
+	map(0x060008, 0x060009).rw(m_acia0, FUNC(acia6850_device::status_r), FUNC(acia6850_device::control_w));
+	map(0x06000a, 0x06000b).rw(m_acia0, FUNC(acia6850_device::data_r), FUNC(acia6850_device::data_w));
+	// octal 0021 - 8-bit
+//	map(0x060010, 0x060011).r(FUNC(blit_state::mouse_buttons)).mask(0x00ff);
+	// octal 0025 - mirror
+//	map(0x060014, 0x060015).r(FUNC(blit_state::mouse_buttons)).mask(0x00ff);
+	// octal 0027 - ???
+	// octal 0030 - 16-bit
+	map(0x060018, 0x060019).w(FUNC(blit_state::start_write));
+	// octal 0040 - 16-bit
+	map(0x060020, 0x060029).ram().share("misccr");
+	// octal 0050 - 16-bit
+//	map(0x060028, 0x060029).w(FUNC(blit_state::init_write));
+	// octal 0060, 0062 - 8-bit - keyboard
+	map(0x060030, 0x060031).rw(m_acia1, FUNC(acia6850_device::status_r), FUNC(acia6850_device::control_w));
+	map(0x060032, 0x060033).rw(m_acia1, FUNC(acia6850_device::data_r), FUNC(acia6850_device::data_w));
+	// octal 0070 - 16-bit
+	map(0x060038, 0x060039).w(FUNC(blit_state::vblank_ack));
+}
+
+static INPUT_PORTS_START(blit)
+INPUT_PORTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( host_rs232_defaults )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_STARTBITS", 0xff, RS232_STARTBITS_1 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_2 )
+	DEVICE_INPUT_DEFAULTS( "FLOW_CONTROL", 0x01, 0x01 )
+DEVICE_INPUT_DEFAULTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( kbd_rs232_defaults )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_STARTBITS", 0xff, RS232_STARTBITS_1 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_2 )
+	DEVICE_INPUT_DEFAULTS( "FLOW_CONTROL", 0x01, 0x01 )
+DEVICE_INPUT_DEFAULTS_END
+
+
+void blit_state::start_write(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	LOG("START <- %04X (addr %06X)\n", data, data << 2);
+	m_videostart = data << 2;
+	m_videoram = (uint8_t *)m_maincpu->space(AS_PROGRAM).get_read_ptr(m_videostart);
```

Comment:
Please don’t use `get_read_ptr`.  It shouldn’t be something you count on.  Also, a byte pointer into a word space is liable to cause Endianness issues.

Reply:
ok - removed get_read_ptr